benchmarks/str-locality.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+import time
+import argparse
+import arkouda as ak
+
+def generate_data(N, seed):
+    prefix = ak.random_strings_uniform(minlen=1, maxlen=16, size=N, seed=seed, characters='numeric')
+    if seed is not None:
+        seed += 1
+    suffix = ak.random_strings_uniform(minlen=1, maxlen=16, size=N, seed=seed, characters='numeric')
+    random_strings = prefix.stick(suffix, delimiter='.')
+    perm = ak.argsort(random_strings.get_lengths())
+    sorted_strings = random_strings[perm]
+    return random_strings, sorted_strings, perm
+
+def time_all_ops(N, trials, seed, correctnessOnly):
```

Comment:
Hmm, for most benchmarks we scale the problem size by the number of locales (`N = N_per_locale * cfg["numLocales"]`), but I'm realizing this test (and substring_search.py/flatten.py do not do that)

Reply:
Another good catch! I fixed it here, but this was actually a copy/paste from `benchmarks/substring_search.py`, so it exists there too. I will start another issue for that.

Reply:
Cool, thanks. Probably need to drop problem size down to `10**7`, at least until optimizations for all these operations go in.

Reply:
Good point. Done.